unified-runtime/test/conformance/lit.cfg.py

```diff
@@ -7,5 +7,68 @@
 
 """
 
-# Disable lit testing (for now)
-config.suffixes = []
+import lit.formats
+import subprocess
+from os import path
+
+
+class NopTestFormat(lit.formats.TestFormat):
+    def getTestsForPath(*args, **kwargs):
+        return
+        yield
+
+    def getTestsInDirectory(*args, **kwargs):
+        return
+        yield
+
+
+config.name = "Unified Runtime Conformance"
+config.test_format = lit.formats.GoogleTest("", "-test")
+config.test_source_root = config.binary_dir
+config.test_exec_root = config.binary_dir
+
+if config.using_l0_v2:
+    config.environment["UR_LOADER_USE_LEVEL_ZERO_V2"] = "1"
+else:
+    config.environment["UR_LOADER_USE_LEVEL_ZERO_V2"] = "0"
+
+if "UR_CTS_ALSO_RUN_KNOWN_FAILURES" in os.environ:
+    config.environment["UR_CTS_ALSO_RUN_KNOWN_FAILURES"] = os.environ[
+        "UR_CTS_ALSO_RUN_KNOWN_FAILURES"
+    ]
+
+if "selector" in lit_config.params:
+    # Specified via passing `-Dselector="backend:*"` to lit
+    config.environment["ONEAPI_DEVICE_SELECTOR"] = lit_config.params["selector"]
+elif os.environ.get("ONEAPI_DEVICE_SELECTOR", ""):
+    lit_config.warning(
+        "ONEAPI_DEVICE_SELECTOR set in environment, using that for UR conformance tests"
+    )
+    config.environment["ONEAPI_DEVICE_SELECTOR"] = os.environ["ONEAPI_DEVICE_SELECTOR"]
+else:
+    ods = ""
+    for a in config.adapters_built:
+        if a != "mock":
+            ods += f"{a}:*;"
+    config.environment["ONEAPI_DEVICE_SELECTOR"] = ods
+
+selector = config.environment["ONEAPI_DEVICE_SELECTOR"]
+if selector == "":
+    lit_config.warning(
+        "No device selector is set (are any adapters enabled?), skipping conformance tests"
+    )
+    config.test_format = NopTestFormat()
```

Comment:
I'm not convinced that changing the lit test format is the way to go when we've been given an empty selector? Why would this happen? What would be the user intent. I suggest we just let it happen and pass off to google test as usual, getting rid of the `NopTestFormat` class in the process

Reply:
It's possible to build UR with no adapters enabled and just run unit tests with the mock adapter. In this case, `check-unified-runtime` should just skip over the conformance tests.

I'm not sure the best way of doing it though, but maybe the `unsupported` flag for ShTest might be better?